hvac/v1/__init__.py

```diff
@@ -39,12 +41,25 @@ def __init__(self, url='http://localhost:8200', token=None,
             'proxies': proxies,
         }
 
+        # auto detect backend version
+        try:
+            json_back = self.list_secret_backends()
+            if 'data' in json_back and 'kv/' in json_back['data']:
+                detected_version = json_back['data']['kv/']['options']['version']
+                self._version = int(detected_version)
+        except:
+            print('Cant determine backend version')
```

Comment:
I'd like to avoid direct calls to print wherever possible. Assuming we simply want to log this behavior (i.e., raising an exception isn't needed), I would vote for use of the logging module. E.g.,:

Add `import logging` and `logger = logging.getLogger(__name__)` to the top-level of this source file. Then this line can be changed to `logger.warning('Cant determine backend version')` or something similar.